addons/mail_client_extension/controllers/main.py

```diff
@@ -182,59 +187,84 @@ def _create_company_from_iap(self, domain):
             values=iap_data,
             subtype_id=request.env.ref('mail.mt_note').id,
         )
-        
+
         return new_company, {'type': 'company_created'}
 
+    def _get_partner_dict(self, partner):
+        return {
+            'id': partner.id,
+            'name': partner.name,
+            'title': partner.function,
+            'email': partner.email,
+            'image': partner.image_128,
+            'phone': partner.phone,
+            'mobile': partner.mobile,
+            'enrichment_info': None,
+        }
+
     @http.route('/mail_client_extension/partner/get', type="json", auth="outlook", cors="*")
-    def res_partner_get_by_email(self, email, name, **kwargs):
+    def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
+
+        assert partner_id or (name and email)
```

Comment:
I don't think "assert" is properly handled by the http client.
Maybe returning a proper structure would be better / more informative for the extension?

Something like:
`return {'error': _('You need to specify at least the partner_id or the name and the email')}`